packages/protoplugin-test/src/custom-options.test.ts

```diff
@@ -0,0 +1,298 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getDescriptorSet } from "./helpers.js";
+import type { DescFile } from "@bufbuild/protobuf";
+import {
+  getCustomOptionBoolean,
+  getCustomOptionFloat,
+  getCustomOptionUint32,
+  getCustomOptionInt32,
+  getCustomOptionInt64,
+  getCustomOptionString,
+  getCustomOptionMessage,
+  getCustomOptionEnum,
+} from "@bufbuild/protoplugin/ecmascript";
+import { proto3 } from "@bufbuild/protobuf";
+
+// Used for custom options of type FooMessage
+const foo = proto3.makeMessageType("FooMessage", [
+  {
+    no: 1,
+    name: "foo",
+    kind: "scalar",
+    T: 5 /* ScalarType.INT32 */,
+  },
+  {
+    no: 2,
+    name: "bar",
+    kind: "scalar",
+    T: 9 /* ScalarType.STRING */,
+  },
+  {
+    no: 3,
+    name: "quux",
+    kind: "scalar",
+    T: 9 /* ScalarType.STRING */,
+  },
+  {
+    no: 4,
+    name: "many",
+    kind: "scalar",
+    repeated: true,
+    T: 9 /* ScalarType.STRING */,
+  },
+  {
+    no: 5,
+    name: "mapping",
+    kind: "map",
+    K: 9 /* ScalarType.STRING */,
+    V: {
+      kind: "scalar",
+      T: 9 /* ScalarType.STRING */,
+    },
+  },
+  {
+    no: 6,
+    name: "unused",
+    kind: "scalar",
+    T: 9 /* ScalarType.STRING */,
+  },
+]);
+
+describe("custom options", function () {
+  let files: DescFile[] = [];
+  beforeEach(() => {
+    files = getDescriptorSet().files.filter(
+      (file) => file.name === "proto/custom_options"
+    );
+  });
```

Comment:
`beforeEach` is good practice to make sure no state is shared across tests, but here it will read the descriptors from disk and parse them for every single test. We have to avoid that to keep the test suite running fast.

Can you switch to `beforeAll`? In my book, it is also an option to just define a `const` in the describe block. Same effect and reads nicer.